lib/oauth/authenticator.js

```diff
@@ -29,13 +29,21 @@ OAuthAuthenticator.prototype.authenticate = function authenticate(req,callback){
   }else{
     throw new Error('authenticate must be called with a request object and callback function');
   }
+
   var authHeader = req && req.headers && req.headers.authorization;
 
   if (authHeader){
     if(authHeader.match(/Bearer/i)){
+      var authenticator = new JwtAuthenticator(self.application);
+
+      if (this.localValidation) {
+        authenticator.withLocalValidation();
+      }
+
       var parts = authHeader.split(' ');
```

Comment:
I actually think that makes it harder to read, since you have to grasp the whole block of code at once. And it's actually two things. First it enables validation (if that is set) then it extracts tokens from a header and tries to authenticate.